chamber/utils/transaction.py

```diff
@@ -12,28 +12,32 @@
 logger = logging.getLogger(__name__)
 
 
-def atomic(func=None):
+def atomic(using=None, savepoint=True):
     """
     Decorator and context manager that overrides django atomic decorator and automatically adds create revision.
     The _atomic closure is required to achieve save ContextDecorator that nest more inner context decorator.
     More here https://stackoverflow.com/questions/45589718/combine-two-context-managers-into-one
     """
 
     @contextmanager
-    def _atomic():
+    def _atomic(using=None, savepoint=True):
         try:
             from reversion.revisions import create_revision
 
-            with transaction.atomic(), create_revision():
+            with transaction.atomic(using, savepoint), create_revision():
                 yield
         except ImportError:
-            with transaction.atomic():
+            with transaction.atomic(using, savepoint):
                 yield
 
-    if func:
-        return _atomic()(func)
+    if callable(using):
+        return _atomic(DEFAULT_DB_ALIAS, savepoint)(using)
```

Comment:
So the `using` is either actually "using" or just a function if used as a decorator? Might be a bit confusing, also if used as a decorator it may be only used on the default db?

Reply:
or is this for backwards compatibility?

Reply:
the reason is that you can use it as a decorator with and without parameters
```
@atomic
def foo():
    pass

@atomic()
def foo():
    pass

@atomic(using='default'):
def foo():
    pass
```

Reply:
I still find it strange, isn't in the last example the `using` argument redeclared both as arg (where the decorated foo function is sent) and kwargs (where 'default' is sent)? I'm probably missing something about decorators.

Reply:
Yes because if you use @atomic without () the first argument of decorator will be the function. Only one argument is using. Maybe help code from django with explanation:

```
def atomic(using=None, savepoint=True):
    # Bare decorator: @atomic -- although the first argument is called
    # `using`, it's actually the function being decorated.
    if callable(using):
        return Atomic(DEFAULT_DB_ALIAS, savepoint)(using)
    # Decorator: @atomic(...) or context manager: with atomic(...): ...
    else:
        return Atomic(using, savepoint)
```

Reply:
thank you